variables.tf

```diff
@@ -2,8 +2,102 @@
+  }
+}
+
+variable "resource_tags" {
+  description = "Optional list of tags to describe the watsonx Assistant instance."
+  type        = list(string)
+  default     = []
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the watsonx Assistant instance created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
+
+  validation {
+    condition = alltrue([
+      for tag in var.access_tags : can(regex("[\\w\\-_\\.]+:[\\w\\-_\\.]+", tag)) && length(tag) <= 128
+    ])
+    error_message = "Tags must match the regular expression \"[\\w\\-_\\.]+:[\\w\\-_\\.]+\", see https://cloud.ibm.com/docs/account?topic=account-tag&interface=ui#limits for more details"
+  }
+}
+
+variable "watsonx_assistant_name" {
+  description = "The name of the watsonx Assistant instance. Required if creating a new instance."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.existing_watsonx_assistant_instance_crn != null || var.watsonx_assistant_name != null
+    error_message = "watsonx Assistant name must be provided when creating a new instance."
+  }
+}
+
+variable "existing_watsonx_assistant_instance_crn" {
+  description = "The CRN of an existing watsonx Assistant instance."
+  type        = string
+  default     = null
+}
+
+variable "existing_watsonx_assistant_instance_name" {
+  description = "The name of an existing watsonx Assistant instance."
+  type        = string
+  default     = null
+}
+
+variable "watsonx_assistant_plan" {
+  description = "The plan that is required to provision the watsonx Assistant instance. For `Trial` and `Lite` accounts, the `watsonx_assistant_service_endpoints` value is ignored and the default service configuration is applied."
+  type        = string
+  default     = null
+
+  validation {
+    condition     = var.existing_watsonx_assistant_instance_crn != null || var.watsonx_assistant_plan != null
+    error_message = "watsonx Assistant plan must be provided when creating a new instance."
+  }
+  validation {
+    condition = anytrue([
+      var.watsonx_assistant_plan == "plus-trial", #  Refers to Trial Account
+      var.watsonx_assistant_plan == "free",       # Refers to Lite account
+      var.watsonx_assistant_plan == "plus",
+      var.watsonx_assistant_plan == "enterprise",
+      var.watsonx_assistant_plan == "enterprisedataisolation",
+    ]) || var.existing_watsonx_assistant_instance_crn != null
+    error_message = "A new watsonx Assistant instance requires a \"Trial\", \"Lite\", \"Plus\", \"Enterprise\", or \"Enterprise with Data Isolation\" plan. [Learn more](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-admin-managing-plan)."
+  }
+}
+
+variable "watsonx_assistant_service_endpoints" {
```

Comment:
can we shorten to just `service_endpoints` ?

Reply:
Done